### PR TITLE
Fix field validation for Cities and Regions on backend 1538

### PR DESCRIPTION
**Pull request by vsynieok**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by Omeliukh-Nazar — APPROVED** [review]

**Review by hryhoriivoloshyn — APPROVED** [review]

**Review by bkondrat — APPROVED** [review]

**Review by tvardero — CHANGES_REQUESTED** [review]
Reviewing

File: EPlast/EPlast.BLL/DTO/Region/RegionDTO.cs [hunk 1]
```diff
@@ -1,26 +1,34 @@
...
     public class RegionDTO
     {
         public int ID { get; set; }
+        [Required, MaxLength(50, ErrorMessage = "Назва округи не має перевищувати 50 символів.")]
         public string RegionName { get; set; }
+        [MaxLength(1000, ErrorMessage = "Опис округи не має перевищувати 1000 символів.")]
         public string Description { get; set; }
         public IEnumerable<RegionAdministrationDTO> Administration { get; set; }
         public IEnumerable<RegionDocumentDTO> Documents { get; set; }
+        [Required, MaxLength(18, ErrorMessage = "Контактний номер округи не може перевищувати 12 символів.")]
```
Comment (tvardero) [inline EPlast/EPlast.BLL/DTO/Region/RegionDTO.cs:16]: Add `[Phone]` attribute to validate that phone format is correct

File: EPlast/EPlast.BLL/DTO/Region/RegionDTO.cs [hunk 2]
```diff
@@ -1,26 +1,34 @@
...
         public string PhoneNumber { get; set; }
+        [Required, MaxLength(50, ErrorMessage = "Місто розташування округи не може перевищувати 50 символів.")]
         public string City { get; set; }
+        [Required, MaxLength(50, ErrorMessage = "Електронна адреса округи не може перевищувати 50 символів.")]
```
Comment (tvardero) [inline EPlast/EPlast.BLL/DTO/Region/RegionDTO.cs:20]: Add `[EmailAddress]` attribute to validate that email format is correct

File: EPlast/EPlast.WebApi/Models/City/CityViewModel.cs [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
 namespace EPlast.WebApi.Models.City
 {
     public class CityViewModel
     {
         public int ID { get; set; }
+        [Required, MaxLength(50, ErrorMessage = "Назва станиці не має перевищувати 50 символів")]
         public string Name { get; set; }
+
+        [StringLength(18, ErrorMessage = "Контактний номер станиці повинен містити 12 цифр")]
         public string PhoneNumber { get; set; }
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Models/City/CityViewModel.cs:15]: `[Phone]`

File: EPlast/EPlast.WebApi/Models/City/CityViewModel.cs [hunk 4]
```diff
@@ -1,17 +1,29 @@
...
+
+        [MaxLength(50, ErrorMessage = "Email станиці не має перевищувати 50 символів")]
         public string Email { get; set; }
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Models/City/CityViewModel.cs:18]: `[EmailAddress]`

File: EPlast/EPlast.WebApi/Models/City/CityViewModel.cs [hunk 5]
```diff
@@ -1,17 +1,29 @@
...
+
+        [MaxLength(256, ErrorMessage = "Посилання на web-сторінку станиці не має перевищувати 256 символів")]
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Models/City/CityViewModel.cs:20]: Remove 256 cap.
Add `[Url]` attribute to validate that Url format is correct.

File: EPlast/EPlast.WebApi/Models/Club/ClubViewModel.cs [hunk 6]
```diff
@@ -1,14 +1,20 @@
...
+using System.ComponentModel.DataAnnotations;
 
 namespace EPlast.WebApi.Models.Club
 {
     public class ClubViewModel
     {
         public int ID { get; set; }
+        [Required, MaxLength(200, ErrorMessage = "Назва куреня не може бути довшою за 200 символів.")]
         public string Name { get; set; }
+        [MaxLength(18, ErrorMessage = "Контактний телефон куреня не може бути довшим за 12 символів.")]
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Models/Club/ClubViewModel.cs:11]: `[Phone]`

File: EPlast/EPlast.WebApi/Models/Club/ClubViewModel.cs [hunk 7]
```diff
@@ -1,14 +1,20 @@
...
         public string PhoneNumber { get; set; }
+        [Required, MaxLength(50, ErrorMessage = "Електронна адрес куреня не може бути довшою за 50 символів.")]
         public string Email { get; set; }
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Models/Club/ClubViewModel.cs:14]: `[EmailAddress]`

File: EPlast/EPlast.BLL/DTO/Region/RegionDTO.cs [hunk 8]
```diff
@@ -1,26 +1,34 @@
...
 using System.Collections.Generic;
-
+using System.ComponentModel.DataAnnotations;
 
 namespace EPlast.BLL.DTO.Region
 {
     public class RegionDTO
     {
         public int ID { get; set; }
+        [Required, MaxLength(50, ErrorMessage = "Назва округи не має перевищувати 50 символів.")]
```
Comment (tvardero) [inline EPlast/EPlast.BLL/DTO/Region/RegionDTO.cs:10]: ErrorMessage:
a) Should not be in any language except for English.
b) Is not used with WebAPI, used only with MVC projects.

Remove all ErrorMessage in every attribute, they are not being used by frontend. ASP.NET will provide default messages in English.

**Review by tvardero — CHANGES_REQUESTED** [review]

**Pull request merged**